pkg/git/v1/checkout.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package git
+package v1
```

Comment:
I reckon name the subpackages for the implementations.

Reply:
Do you mean naming the package go-git instead of v1? Would that not create confusion as there would be two packages with the same name.